sklearn/cluster/_affinity_propagation.py

```diff
@@ -523,7 +523,9 @@ def predict(self, X):
 
         if self.cluster_centers_.shape[0] > 0:
             with config_context(assume_finite=True):
-                return pairwise_distances_argmin(X, self.cluster_centers_)
+                return pairwise_distances_argmin(
+                    X, self.cluster_centers_, metric="fast_euclidean"
```

Comment:
To be extra explicit we select the fastest variant (we don't care about the distance values for this method):

```suggestion
                    X, self.cluster_centers_, metric="fast_sqeuclidean"
```

Reply:
This deserves a short comment in the code.

Reply:
That's right. I specified `"fast_euclidean"` knowingly that the user should be knowledgeable about the trick.
Internally, `"fast_sqeuclidean"` will be used in any case if distances aren't returned, I do not have strong preference.

As mentioned by @lorentzenchr in his review, I will add a comment here giving the motive.

(IIRC, we already have had this discussion in #20254 but this got lost in previous threads.)